app/src/main/java/com/google/android/sambadocumentsprovider/browsing/broadcast/BroadcastUtils.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.google.android.sambadocumentsprovider.browsing.broadcast;
+
+import android.util.Log;
+
+import com.google.android.sambadocumentsprovider.browsing.BrowsingException;
+
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+class BroadcastUtils {
+  private static final int FILE_SERVER_NODE_TYPE = 0x20;
+
+  /**
+   * Generates a NetBIOS name query request.
+   * https://tools.ietf.org/html/rfc1002
+   * Section 4.2.12
+   */
+  static byte[] createPacket() {
+    ByteBuffer os = ByteBuffer.allocate(50);
```

Comment:
Maybe be explicit by calling order(ByteOrder.BIG_ENDIAN) even though it looks like it is the default. rfc1002 references rfc 990 - https://tools.ietf.org/html/rfc990 which defines the byte order in the other rfc.